### PR TITLE
[ENH] make the number of vertices the first dimension of image data

### LINKED ISSUE #4569: What should be the shape of data in SurfaceImage?
Not sure what makes more sense: to have data as 
- n_vertices * n_timepoints (current)
- or n_timepoints * n_vertices (what this change would do)?

The change was required to fix loading data from a real life fmriprep dataset: https://github.com/nilearn/nilearn/pull/4507#issuecomment-2297087087

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/4507#discussion_r1722167540_
            
See discussion in source PR for more info

### PR DESCRIPTION
**Pull request by jeromedockes**
- fixes #4569 
- supersedes #4576 

as discussed with @Remi-Gau

### CODE REVIEW

File: nilearn/experimental/surface/_surface_image.py
Comment (bthirion) [inline nilearn/experimental/surface/_surface_image.py:193]: Also give the number of edges and faces ?

File: nilearn/experimental/surface/maskers.py
Comment (bthirion) [inline nilearn/experimental/surface/maskers.py:496]: Sorry for my lack of culture, but why not 
```
>>> unmasked = masker.inverse_transform(masked)
```

**Comment by bthirion** [comment]
Do you need some help on that one ?

**Comment by Remi-Gau** [comment]
Yes
At the moment I was trying to resolve merge conflicts and update some tests because so many things moved around.
But if someone has the bandwidth to work on this, I would like that: especially that my numpy is defo not as proficient as that of some people in the team.

**Comment by jeromedockes** [comment]
I had understood from our last meeting that it had to wait for you to merge another pr first, but if that is done I can have a look at this one

**Comment by Remi-Gau** [comment]
Yeah technically this is still blocked by the surface label masker pr.
But this one should hopefully be merged on Monday so after that I will fix the remaining  merge conflict and hand it back to you.
OK with you?

**Comment by jeromedockes** [comment]
> Yeah technically this is still blocked by the surface label masker pr.
> But this one should hopefully be merged on Monday so after that I will fix the remaining  merge conflict and hand it back to you.

sounds good, thanks! again, I can take care of the conflicts -- it's my own fault for taking too long to complete the PR :D

**Comment by Remi-Gau** [comment]
- blocked by #4771

**Comment by Remi-Gau** [comment]
@jeromedockes 
Can I let you take over this one?
I think you will be faster at finishing this and I still need to take care of the second level stuff.

**Comment by jeromedockes** [comment]
> @jeromedockes 
> Can I let you take over this one?

ok :)

**Comment by Remi-Gau** [comment]
don't hesitate to undo some of the 'fixes' I did.

**Review by bthirion — APPROVED** [review]
LGTM

**Comment by Remi-Gau** [comment]
[[full doc]](https://github.com/nilearn/nilearn/pull/4591/ commits/973ff5537f4fa6f0992de5db3498a8b6ba9cbf9e) failed because of an unrelated issue.

should be fixed: asking for another full build again

File: examples/07_advanced/plot_surface_image_and_maskers.py
Comment (Remi-Gau) [inline examples/07_advanced/plot_surface_image_and_maskers.py:1]: @jeromedockes 
I think there is one remaining issue in this example

Comment (Remi-Gau) [reply examples/07_advanced/plot_surface_image_and_maskers.py:1]: Will fix in the follow up PR

**Review by Remi-Gau — APPROVED** [review]

**Pull request merged**
